lib/write-type.js

```diff
@@ -31,6 +31,7 @@ function getWriteType(options) {
   var bin = binarraybuffer ? bin_arraybuffer : bin_buffer;
   var usemap = HAS_MAP && options && options.usemap;
   var map = usemap ? map_to_map : obj_to_map;
+  var strip_undef = options && options.strip_undef;
```

Comment:
how about calling it `omit_undefined_properties`? 
`undef` is needlessly terse, and by spelling out the "properties" part, we're clear that it doesn't remove them from array elements.